Creating `bootstrap.lis` breaks compilation when using enum

This is because enum variants constructors are written to `bootstrap.go`. Maybe it's better to just write them in the same files where enums are defined?

Reproduction:

```rs
// main.lis
fn main() {
  let _ = A.A
}

enum A {
  A,
}

// bootstrap.lis
// empty
```
running:
```bash
# test
./main.go:6:6: undefined: MakeAA
```

--- a/crates/emit/src/collectors.rs
+++ b/crates/emit/src/collectors.rs
@@ -1,3 +1,5 @@
+use rustc_hash::FxHashMap as HashMap;
+
 use super::names::go_name;
 use crate::Emitter;
 use syntax::program::{Definition, File};
@@ -181,15 +183,12 @@ impl Emitter<'_> {
         }
     }
 
-    /// Register make function names for all enums and generate code for current module's enums.
-    ///
-    /// Combines the previous two-phase pattern (register names, then generate code) into a
-    /// single pass over definitions.
-    pub(crate) fn collect_make_functions(&mut self) -> Vec<String> {
+    /// Register make function names for all enums; return bodies keyed by declaring file_id.
+    pub(crate) fn collect_make_functions(&mut self) -> HashMap<u32, Vec<String>> {
         self.register_prelude_make_functions();
 
         let module_prefix = format!("{}.", self.current_module);
-        let mut code = Vec::new();
+        let mut code: HashMap<u32, Vec<String>> = HashMap::default();
 
         // Collect enum info first to avoid borrow conflicts
         let enums: Vec<_> = self
@@ -197,22 +196,33 @@ impl Emitter<'_> {
             .definitions
             .iter()
             .filter_map(|(key, definition)| {
-                if let Definition::Enum { name, variants, .. } = definition {
+                if let Definition::Enum {
+                    name,
+                    variants,
+                    name_span,
+                    ..
+                } = definition
+                {
                     if name == "Option" || name == "Result" || name == "Partial" {
                         return None;
                     }
-                    Some((key.to_string(), name.clone(), variants.clone()))
+                    Some((
+                        key.to_string(),
+                        name.clone(),
+                        variants.clone(),
+                        name_span.file_id,
+                    ))
                 } else {
                     None
                 }
             })
             .collect();
 
-        for (_, name, variants) in &enums {
+        for (_, name, variants, _) in &enums {
             self.register_make_functions(name, variants);
         }
 
-        for (key, _name, variants) in &enums {
+        for (key, _name, variants, file_id) in &enums {
             if !key.starts_with(&module_prefix) {
                 continue;
             }
@@ -221,7 +231,8 @@ impl Emitter<'_> {
                 continue;
             }
             for variant in variants {
-                code.push(self.create_make_function_code(key, &variant.name));
+                let fn_code = self.create_make_function_code(key, &variant.name);
+                code.entry(*file_id).or_default().push(fn_code);
             }
         }
 

--- a/crates/emit/src/lib.rs
+++ b/crates/emit/src/lib.rs
@@ -470,10 +470,9 @@ impl<'a> Emitter<'a> {
         self.collect_exported_method_names(files);
         self.collect_impl_bounds(files);
         self.collect_enum_layouts();
-        let make_functions = self.collect_make_functions();
+        let mut make_functions_by_file = self.collect_make_functions();
 
         let mut output_files = Vec::new();
-        let should_create_bootstrap = files.len() > 1 && !make_functions.is_empty();
 
         let package_name = if module_id == self.ctx.entry_module {
             "main".to_string()
@@ -482,47 +481,12 @@ impl<'a> Emitter<'a> {
             go_name::sanitize_package_name(raw).into_owned()
         };
 
-        if should_create_bootstrap {
-            let mut bootstrap_source = OutputCollector::new();
-            for function in &make_functions {
-                bootstrap_source.collect_with_blank(function.clone());
-            }
-
-            let bootstrap_source_str = bootstrap_source.render();
-
-            let unused_imports =
-                Self::unused_imports_for_current_module(self.ctx.unused, &self.current_module);
-            let mut import_builder = ImportBuilder::new(
-                &self.ctx.go_module,
-                unused_imports,
-                self.ctx.go_package_names,
-            );
-
-            // Collect imports from all files in the module to get aliased imports
-            for file in files {
-                import_builder.collect_from_file(file);
-            }
-
-            import_builder.extend_with_modules(&self.ensure_imported);
-            if self.flags.needs_stdlib {
-                import_builder.require_stdlib();
-            }
-            import_builder.filter_unreferenced(&bootstrap_source_str);
-
-            output_files.push(OutputFile {
-                name: "bootstrap.go".to_string(),
-                imports: import_builder.build(),
-                source: bootstrap_source_str,
-                package_name: package_name.clone(),
-            });
-        }
-
         for file in files {
             let mut source = OutputCollector::new();
 
-            if !should_create_bootstrap {
-                for function in &make_functions {
-                    source.collect_with_blank(function.clone());
+            if let Some(functions) = make_functions_by_file.remove(&file.id) {
+                for function in functions {
+                    source.collect_with_blank(function);
                 }
             }
 

--- a/tests/spec/build/mod.rs
+++ b/tests/spec/build/mod.rs
@@ -3920,3 +3920,26 @@ fn main() {
 
     assert_build_snapshot!(fs, "github.com/user/myproject");
 }
+
+#[test]
+fn user_file_named_bootstrap_does_not_collide() {
+    let mut fs = MockFileSystem::new();
+
+    fs.add_file(
+        ENTRY_MODULE_ID,
+        "main.lis",
+        r#"
+fn main() {
+  let _ = A.A
+}
+
+enum A {
+  A,
+}
+"#,
+    );
+
+    fs.add_file(ENTRY_MODULE_ID, "bootstrap.lis", "");
+
+    assert_build_snapshot!(fs, "github.com/user/myproject");
+}

